AOC3/src/main/kotlin/Day16.kt

```diff
@@ -15,19 +15,16 @@ fun main() {
     while (i <= iterations) {
         orders.forEach { order ->
             when (order[0]) {
-                's' -> spinDanceMove(Integer.parseInt(order.slice(1 until order.length)))
+                's' -> spinDanceMove(order.drop(1).toInt())
                 'x' -> {

```

Comment:
There are quite a few String processing methods in Kotlin that make your approach here much more readable and efficient.